Revolution Rotation Calculator.py

```diff
@@ -78,93 +75,79 @@ def AbilityRotation(Permutation, AttackSpeed, Activate_Bleeds, Gain, Start_Adren
                     Current_Buff = AdjustCooldowns(Current_Buff, Adrenaline, AbilityTime[ability]) # Will also manage cooldowns
                     break
-        # --- Determines whether thresholds or ultimates may be used --- # 
+        # --- Determines whether thresholds or ultimates may be used --- #
         if Clock < Time:
-            if Adrenaline >= 50:
-                for Ability in AbilityType:
-                    if AbilityType[Ability] == 'T':
-                        if Ability not in TrackCooldown:
-                            Ready[Ability] = True
-                    if Adrenaline == 100:
-                        for Ability in AbilityType:
-                            if AbilityType[Ability] == 'U':
-                                if Ability not in TrackCooldown:
-                                    Ready[Ability] = True
-            if Adrenaline < 50:
-                for Ability in AbilityType:
-                    if AbilityType[Ability] == 'T':
-                        Ready[Ability] = False
+            if Adrenaline == 100:
+                for Ability in (a for a in UltimateIterator if a not in TrackCooldown):
+                    Ready[Ability] = True
+                for Ability in (a for a in ThresholdIterator if a not in TrackCooldown):
+                    Ready[Ability] = True
+            elif Adrenaline >= 50:
+                for Ability in (a for a in ThresholdIterator if a not in TrackCooldown):
+                    Ready[Ability] = True
+            elif Adrenaline < 50:
+                for Ability in ThresholdIterator:
+                    Ready[Ability] = False
             if Adrenaline != 100:
-                for Ability in AbilityType:
-                    if AbilityType[Ability] == 'U':
-                        Ready[Ability] = False
+                for Ability in UltimateIterator:
+                    Ready[Ability] = False
         # --- Determines if any abilities available/ whether auto attacks must be used --- #
         if Clock < Time:
             AbilityAvailable = False
-            for ability in Permutation:
-                if Ready[ability] == True:
-                    AbilityAvailable = True
-            if AbilityAvailable == False:
-                if Auto_Available() == True:
+            for _ in (a for a in Permutation if Ready[a]):
+                AbilityAvailable = True
+                break
+            if AbilityAvailable is False:
+                if Auto_Available() is True:
                     if (Clock + AttackSpeedCooldowns[AttackSpeed]) <= Time:
                         Clock +=  AttackSpeedCooldowns[AttackSpeed]
                         Clock = round(Clock, 1)
                     else:
                         Clock += (Time - Clock)
-                        Clock = round(Clock, 1)
+                        Clock = round(Clock, 1) # PyCharm is complaining that this is never used..

```

Comment:
PyCharm is complaining that this is never used..

Reply:
The round Clock was because I kept getting issues where 1.8 + 1.8 = 3.5999999999 (I wasn't sure what was going on so I just rounded Clock)